### ISSUE TITLE
Question about user agent (lower-case)

### ISSUE THREAD

**Issue by eurobank** [CIDRAM/CIDRAM#400]
What exactly does the "user agent - lower-case" means/does?

**Comment by mikeruss1** [comment]
you put the UA you are going to match against in all lower case to get a match

**Comment by eurobank** [comment]
@mikeruss1 Not sure i understand. 

If the agent is "Velen" and i want to block at once "Velen" and "velen" what must i use?

**Comment by Maikuolan** [comment]
When using "user agent lowercase" as the condition's source, the condition's value would need to be written as lowercase to be able to match anything (having uppercase characters would cause the condition to never match anything), but would otherwise be capable of matching the cited user agent regardless of the case it uses (e.g., whether "Velen", "velen", "vELEN", "VeLeN", "vElEn", etc, it wouldn't matter). Effectively, it provides a way to treat and match user agents in a case-insensitive way, but does so by "normalising" a copy of the user agent to lowercase.

In other words: When the case *is* significant (i.e., you *do* want to match things like "Bad User Agent", but specifically *don't* want to match things like "bAd uSer aGent"), don't use "user agent lowercase" as the condition's source; just use the normal "user agent" condition source. But, when case *isn't* significant (i.e., you want to match any/all case permutations of the user agent in question), use "user agent lowercase" as the condition's source, and ensure the condition's value is lowercase, and it should be able to match any/all case permutations.

**Comment by Maikuolan** [comment]
But, forewarning, there's a caveat to the normalisation process: It relies on PHP's [`strtolower()`](https://www.php.net/strtolower) function, and that function isn't Unicode-aware. What that means is, although it'll work as expected for ANSI characters (i.e., A through Z), it'll produce unexpected results for non-ANSI characters.

As an example, attempting to convert these into lowercase:
```
Привет!
Γειά σου!
Բարեւ Ձեզ!
```

What we would normally expect:
```
привет!
γειά σου!
բարեւ ձեզ!
```

What we would actually get instead:
```
ð¿ривет!
î³ειά σου!
õ¢արեւ ձեզ!
```

There are some other ways to deal with those situations, so it's not exactly a big a problem, but something to be aware of nonetheless.

**Comment by eurobank** [comment]
thanks guys.
